BSP/STM32F413/Src/BSP_I2C.c

```diff
@@ -45,282 +78,310 @@ void BSP_I2C_Init(void) {
                                                                 // Note I left shifted the address. That's because the address is supposed
                                                                 // to be 7-bits and the least significant bit indicates the R/W setting
 	I2C_Init(I2C3, &I2C_InitStruct);
+	I2C_ITConfig(I2C3, I2C_IT_EVT, ENABLE); //Enable interrupts
 	I2C_Cmd(I2C3, ENABLE);
+	#ifdef RTOS
+	OSQCreate(&I2Crx, "I2C RX", AMT, &err);
+	OSQCreate(&I2Ctx, "I2C TX", AMT, &err);
+	#endif
 }
-
-/**
- * @brief   Transmits data onto the I2C bus.
- * @param   deviceAddr : the device/IC that the data is intended for.
- * @param   regAddr : the register address to write to in the IC's memory.
- * @param   txData : the data array to be sent onto the bus.
- * @param   txLen : the length of the data array.
- * @return  error status, 0 if fail, 1 if success
- */
-uint8_t BSP_I2C_Write(uint8_t deviceAddr, uint16_t regAddr, uint8_t *txData, uint32_t txLen) {
-    int timeout_count = 0;
-	while(I2C_GetFlagStatus(I2C3, I2C_FLAG_BUSY)){
-		// Assume running at 80 MHz
-		timeout_count++;
-		// Returns and breaks after timeout threshold
-		if(timeout_count > TIMEOUT_THRESHOLD) {
-			return ERROR;
-		}
+/*****THE FOLLOWING CODE IS FOR THE BARE METAL VERSION OF THE BPS*****/
+#ifdef BAREMETAL
+void I2C3_EV_IRQHandler(void){
```

Comment:
This is a pretty long ISR

Reply:
This is what it is called in the external interrupt vector table. I thought that is what I was supposed to use. It's in startup_stm32f413xx.s line 231.